testSuite/shared/src/test/scala/testsuite/javalib/text/DecimalFormatTest.scala

```diff
@@ -1,14 +1,663 @@
+    val df = NumberFormat.getIntegerInstance.asInstanceOf[DecimalFormat]
+    df.setGroupingSize(4)
+    assertEquals("123", "123", df.format(123))
+    assertEquals("0", "0", df.format(0))
+    assertEquals("-123", "-123", df.format(-123))
+    assertEquals("1000", "1000", df.format(1000))
+    assertEquals("10,0000", "10,0000", df.format(100000))
+    assertEquals("1000,0000", "1000,0000", df.format(10000000))
+    assertEquals("21,4748,3647", "21,4748,3647", df.format(Int.MaxValue))
+    assertEquals("922,3372,0368,5477,5807", "922,3372,0368,5477,5807", df.format(Long.MaxValue))
+    assertEquals("-1000", "-1000", df.format(-1000))
+    assertEquals("-10,0000", "-10,0000", df.format(-100000))
+    assertEquals("-1000,0000", "-1000,0000", df.format(-10000000))
+    assertEquals("-21,4748,3648", "-21,4748,3648", df.format(Int.MinValue))
+    assertEquals("-922,3372,0368,5477,5808", "-922,3372,0368,5477,5808", df.format(Long.MinValue))
+  }
+
+
+  @Test def test_format_decimals(): Unit = {
+    val df = NumberFormat.getNumberInstance.asInstanceOf[DecimalFormat]
+    assertEquals("123.457", "123.457", df.format(123.4567))
+    assertEquals("0.0", "0", df.format(0.0))
+    assertEquals("-123.457", "-123.457", df.format(-123.4567))
+    assertEquals("1,000.123", "1,000.123", df.format(1000.1234))
+    assertEquals("10,000.123", "10,000.123", df.format(10000.1234))
+    assertEquals("10,000,000.123", "10,000,000.123", df.format(10000000.1234))
+//    assertEquals("21,4748,3647", "21,4748,3647", df.format(Int.MaxValue))
+//    assertEquals("922,3372,0368,5477,5807", "922,3372,0368,5477,5807", df.format(Long.MaxValue))
+    assertEquals("-1,000.123", "-1,000.123", df.format(-1000.1234))
+    assertEquals("-10,000.123", "-10,000.123", df.format(-10000.1234))
+    assertEquals("-10,000,0000.123", "-10,000,000.123", df.format(-10000000.1234))
+//    assertEquals("-21,4748,3648", "-21,4748,3648", df.format(Int.MinValue))
+//    assertEquals("-922,3372,0368,5477,5808", "-922,3372,0368,5477,5808", df.format(Long.MinValue))
+  }
+
+
+  @Test def test_format_decimals_min_fractions(): Unit = {
+    val df = NumberFormat.getNumberInstance.asInstanceOf[DecimalFormat]
+    df.setMinimumFractionDigits(5)
+    assertEquals("123", "123.00000", df.format(123))
+    assertEquals("123.45670", "123.45670", df.format(123.4567))
+    assertEquals("123.00010", "123.00010", df.format(123.0001))
+    assertEquals("0.0", "0.00000", df.format(0.0))
+    assertEquals("-123", "-123.00000", df.format(-123))
+    assertEquals("-123.45670", "-123.45670", df.format(-123.4567))
+    assertEquals("-123.00010", "-123.00010", df.format(-123.0001))
+    assertEquals("1,000.12340", "1,000.12340", df.format(1000.1234))
+    assertEquals("10,000.12340", "10,000.12340", df.format(10000.1234))
+    assertEquals("10,000,000.12340", "10,000,000.12340", df.format(10000000.1234))
+    //    assertEquals("21,4748,3647", "21,4748,3647", df.format(Int.MaxValue))
```

Comment:
Do you think your new algorithm can handle these cases, where the `MaxValue * multpiler` overflows?